src/color.rs

```diff
@@ -386,47 +387,84 @@ fn clamp_f32(val: f32) -> u8 {
 
 #[inline]
 fn parse_color_function(name: &str, arguments: &mut Parser) -> Result<Color, ()> {
-    let (is_rgb, has_alpha) = match_ignore_ascii_case! { name,
-        "rgba" => (true, true),
-        "rgb" => (true, false),
-        "hsl" => (false, false),
-        "hsla" => (false, true),
+    let is_rgb = match_ignore_ascii_case! { name,
+        "rgba" => true,
+        "rgb" => true,
+        "hsl" => false,
+        "hsla" => false,
         _ => return Err(())
     };
 
     let red: u8;
     let green: u8;
     let blue: u8;
+    let mut uses_commas = false;
     if is_rgb {
```

Comment:
I believe that, now that this function is basically a big:

```
if is_rgb {
    // ...
} else {
    // ...
}

try_parse_alpha()
```

we can split it the giant `if` into two functions to get the relevant rgb components:

```rust
fn parse_rgb_components_rgb(arguments: &mut Parser) -> Result<(u8, u8, u8, bool), ()>;
fn parse_rgb_components_hsl(arguments: &mut Parser) -> Result<(u8, u8, u8, bool), ()>;
```

This would look something like:

```rust
let (red, green, blue, uses_commas) = if is_rgb {
    parse_rgb_components_rgb(arguments)?
} else {
    parse_rgb_components_hsl(arguments)?
};

// Code for parsing the optional alpha value.
```

What do you think? I believe this should make this function much easier to read and understand.